### PR TITLE
Auth refactor

### PR DESCRIPTION
**Pull request by evansiroky**
Add a server-side route for auth0's silent authentication.

This PR is required in order for https://github.com/catalogueglobal/datatools-ui/pull/12 to work properly.

### CODE REVIEW

**Comment by abyrd** [comment]
@evansiroky will merging this break any backward compatibility? It doesn't look like it, it just adds a new endpoint right?

File: src/main/java/com/conveyal/datatools/manager/DataManager.java [hunk 1]
```diff
@@ -199,6 +199,16 @@ private static void registerRoutes() throws IOException {
             return null;
         });
 
+        InputStream auth0Stream = DataManager.class.getResourceAsStream("/public/auth0-silent-callback.html");
+        String auth0html = IOUtils.toString(auth0Stream);
```
Comment (abyrd) [inline src/main/java/com/conveyal/datatools/manager/DataManager.java]: this field should probably be final and maybe static.

File: src/main/java/com/conveyal/datatools/manager/DataManager.java [hunk 2]
```diff
@@ -200,7 +200,7 @@ private static void registerRoutes() throws IOException {
         });
 
         InputStream auth0Stream = DataManager.class.getResourceAsStream("/public/auth0-silent-callback.html");
-        String auth0html = IOUtils.toString(auth0Stream);
+        final String auth0html = IOUtils.toString(auth0Stream);
```
Comment (evansiroky) [inline src/main/java/com/conveyal/datatools/manager/DataManager.java]: Is this what you had in mind, @abyrd ?

**Comment by evansiroky** [comment]
This just adds a new endpoint and doesn't change anything else.

File: src/main/java/com/conveyal/datatools/manager/DataManager.java (hunk 2 above)
Comment (abyrd) [inline src/main/java/com/conveyal/datatools/manager/DataManager.java]: I just meant that as a String that is predefined and hardwired and will not change between instances of the class (it's a constant), the convention is to make it static and final. Adding the final keyword is all you can do in this context/scope inside a method, although by that line of reasoning we could/should add many other final keywords all over the place. But usually what you'd do is pull a constant field like this out to the outermost level of the class and declare it `static final`, with an all caps identifier. This is just Java convention for a symbolic constant.

**Comment by landonreed** [comment]
LGTM

**Pull request merged**